src/main/java/org/jboss/staxmapper/XMLExtendedStreamReaderImpl.java

```diff
@@ -451,9 +451,9 @@ public long getLongAttributeValue(final int index) throws XMLStreamException {
                 throw intParseException(e, getLocation());
             }
         } else if (kind.isEnum()) {
-            final List<T> list = new ArrayList<T>();
+            final List<T> list = new ArrayList<>();
             for (String s : Spliterable.over(getAttributeValue(index), ' ')) {
-                list.add(kind.cast(Enum.valueOf(kind.asSubclass(Enum.class), s)));
+                list.add((T) kind.cast(Enum.valueOf(kind.asSubclass(Enum.class), s)));
```

Comment:
I was looking wrongly, this change is still here.
and you are right output of kind.cast is T, but looks like this is compiler error.
- IDE doesn't complain, 
- using jdk7 work
- using jdk8 with target 7 works
- using jdk8u102 with target 8, complains without explicit cast.
- using jdk9EAu138 with target 8, works

so it looks like bug in JDK8 compiler, noting else.